cpp/modules/deck.gl/core/src/arrow/row.h

```diff
@@ -43,18 +55,44 @@ class Row {
   auto getBool(const std::string& columnName, bool defaultValue = false) const -> bool;
   auto getString(const std::string& columnName, const std::string& defaultValue = "") const -> std::string;
 
-  auto getFloatVector2(const std::string& columnName, const mathgl::Vector2<float>& defaultValue = {}) const
-      -> mathgl::Vector2<float>;
-  auto getDoubleVector2(const std::string& columnName, const mathgl::Vector2<double>& defaultValue = {}) const
-      -> mathgl::Vector2<double>;
-  auto getFloatVector3(const std::string& columnName, const mathgl::Vector3<float>& defaultValue = {}) const
-      -> mathgl::Vector3<float>;
-  auto getDoubleVector3(const std::string& columnName, const mathgl::Vector3<double>& defaultValue = {}) const
-      -> mathgl::Vector3<double>;
-  auto getFloatVector4(const std::string& columnName, const mathgl::Vector4<float>& defaultValue = {}) const
-      -> mathgl::Vector4<float>;
-  auto getDoubleVector4(const std::string& columnName, const mathgl::Vector4<double>& defaultValue = {}) const
-      -> mathgl::Vector4<double>;
+  template <typename T>
+  auto getVector2(const std::string& columnName, const mathgl::Vector2<T>& defaultValue = {}) const
+      -> mathgl::Vector2<T> {
+    auto optionalData = this->_getVectorData<T>(columnName);
+    if (!optionalData) {
+      probegl::WarningLog() << "Unsupported vector data type, returning default value";
+      return defaultValue;
+    }
+
+    auto data = optionalData.value();
+    return mathgl::Vector2<T>{data.size() > 0 ? data.at(0) : 0, data.size() > 1 ? data.at(1) : 0};
+  }
+  template <typename T>
+  auto getVector3(const std::string& columnName, const mathgl::Vector3<T>& defaultValue = {}) const
+      -> mathgl::Vector3<T> {
+    auto optionalData = this->_getVectorData<T>(columnName);
+    if (!optionalData) {
+      probegl::WarningLog() << "Unsupported vector data type, returning default value";
```

Comment:
Be careful with these warnings, these could cause massive cascades if processing a million row table.

The probe.gl logging system in JS has warning deduplication as a built-in feature. I.e. each warning message string is cached and only printed once.

Reply:
I could put that in place if it's not too much work, not sure how you went about doing that in JS. For now, I think there should be something in place that notifies the user of the API that something has gone wrong during the conversion though, as ugly as this may be

Reply:
Just a `set` recording which strings have already been logged. Could also be a map<string, timestamp> and only log if say 1 second has passed.

I believe the log function has an `option.once`. This is set by default on log.warn and log.error (the api is slightly different).

Reply:
Opened up a new issue #129